changelogger: Recognize more non-feature-branch names

anomiex opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
We were already recognizing 'main', 'master', and 'trunk'. Let's take
some more from those Composer itself recognizes: 'current', 'default',
'develop', 'latest', 'next', 'support', and 'tip'.

And let's throw 'production' in there too, I've seen that one somewhere.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
None

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Try creating a branch with any of these names, then do `changelogger add`. It should suggest a date-based filename instead of using the branch name.
* Any other names people have been using?

coder-karen reviewed (commented):
Attempted to give this a test but I'm possibly missing something. Both creating a new branch off 'master' (eg. 'trunk') and editing the changed files, and creating a 'trunk' branch off this PR branch, then running 'jetpack changelog add' or 'jetpack changelogger add', I'm not being prevented from adding a changelog entry (I'm not running 'changelogger add' on it's own though).

anomiex commented:
It will never prevent you. The difference this makes is this:
```
$ jetpack changelog add plugins/jetpack
When writing your changelog entry, please use the format 'Subject: change description.'
Here is an example of a good changelog entry:
  Sitemaps: ensure that the Home URL is slashed on subdirectory websites.

Name your change file [default: add-changelogger-more-non-feature-branch-names] >
```
See the default there is based on the branch name.
```
$ git checkout -b trunk
Switched to a new branch 'trunk'
Files within the packages directory have changed. Consider running: jetpack install --all
$ jetpack changelog add plugins/jetpack
When writing your changelog entry, please use the format 'Subject: change description.'
Here is an example of a good changelog entry:
  Sitemaps: ensure that the Home URL is slashed on subdirectory websites.

Name your change file [default: 2022-06-10-13-38-02-096552] >
```
There it recognized the "trunk" branch as special and offered a date-based default filename instead. That happened for "trunk" before this PR, although the code was refactored so testing again isn't a bad idea.

The change is that we now have some other branches recognized too.
```
$ git checkout -b production
Switched to a new branch 'production'
Files within the packages directory have changed. Consider running: jetpack install --all
$ jetpack changelog add plugins/jetpack
When writing your changelog entry, please use the format 'Subject: change description.'
Here is an example of a good changelog entry:
  Sitemaps: ensure that the Home URL is slashed on subdirectory websites.

Name your change file [default: 2022-06-10-13-37-51-840326] >
```

coder-karen reviewed (approved):
I completely misunderstood. Thanks for explaining - it definitely works as expected in that case.

> Any other names people have been using?

Not so sure about that, but what if someone were to use 'release'? Anyway, will approve this.
